tests/run-tests.py

```diff
@@ -974,7 +984,15 @@ def main():
     else:
         raise ValueError("target must be one of %s" % ", ".join(LOCAL_TARGETS + EXTERNAL_TARGETS))
 
-    if len(args.files) == 0:
+    if args.run_failed:
```

Comment:
If `args.files` is specified, should that override `args.run_failed`?  I guess it doesn't really matter, but maybe for consistency `args.files` should take preference?

Reply:
With the wording 'run only failed' I'd say it takes precedence over passing files, but then indeed it's not clear that passing files as well will simply ignore them. But the same problem arises when doing it the other way around, then run_failed would silently be ignored. So if files should take precedence then either I change the doc on run-failures to state that it won't have effect if files are passed, or make it a hard error i.e. raise an ArgumentError (or something like that) if both are used?

Reply:
Maybe just hard-fail if you pass both this arg and explicit files?

Reply:
Ok, seems most reasonable